uavcan/node/430.GetInfo.0.1.uavcan

```diff
@@ -0,0 +1,69 @@
+#
+# Full node info request.
+# All of the returned information must be static (unchanged) while the node is running, excepting the Status field.
+# It is highly recommended to support this service on all nodes.
+#
+
+---
+
+# The current node status (most recent heartbeat message).
+# This is the same value that is periodically published by the node.
+# The objective of its inclusion here is to ensure that the node did not restart since the last time
+# this message was received from it, because a restart may invalidate the node information.
+Heartbeat.1 most_recent_heartbeat
+
+# The UAVCAN protocol version implemented on this node, both major and minor.
+# Not to be changed while the node is running.
+Version.1 protocol_version
+
+# The version information must not be changed while the node is running.
+# The correct hardware version must be reported at all times, excepting software-only nodes, in which
+# case it should be set to zeros.
+# If the node is equipped with a UAVCAN-capable bootloader, the bootloader should report the software
+# version of the installed application, if there is any; if no application is found, zeros should be reported.
+Version.1 hardware_version
+Version.1 software_version
+
+# A version control system (VCS) revision number or hash. Not to be changed while the node is runnning.
+# For example, this field can be used for reporting the short git commit hash of the current
+# software revision.
+# Set to zero if not used.
+uint32 software_vcs_revision_id
```

Comment:
```suggestion
uint64 software_vcs_revision_id
```
for future-proofing.

Reply:
For that to make sense, there should be either:

- More than 2^32 commits.
- The number of commits should be large enough for a short hash collision to be likely. Consider this: assuming that the short hash is perfect, you need ~3000 commits to reach the collision probability of 0.1% with a 32-bit hash. 3k commits is not much (for reference, there are 28k commits in the PX4 Firmware repository), but you also have the major and the minor version numbers to further narrow down the exact revision, so your real constraint is that there shouldn't be more than a couple thousand of commits sharing the same minor version number. Which seems sensible.

Do we really want 64 bits here?

Reply:
I agree with @thirtytwobits here, I would prefer 64 bits of vcs commit it, especially since since GetInfo is not going to be called very often anyway.

Reply:
@antoinealb what is the rationale?

Reply:
If I computed correctly, at 28k commit, you have about 9% chance of a collision in the 32 bit git hash. Of course like you said it can be narrowed down using major and minor, but that's a hassle that is really easy to avoid. If we use 64 bits instead for example, the probability drops to about 10e-11.

Reply:
> If we use 64 bits instead for example, the probability drops to about 10e-11.

Okay. That is probably an overkill, how about we used 40-bit here? For 40-bit hash:

* ~0.1% at 50k commits
* ~24% at 784k commits (that's how many there are in the Linux kernel currently)

For 48-bit hash:

* ~0.0004% at 50k commits
* ~0.1% at 784k commits

Reply:
40 or 48 are ok for me.

Reply:
Done, increased to 40 bit.